src/components/SpecTestContainer.tsx

```diff
@@ -17,54 +17,100 @@ import TextField from 'material-ui/TextField';
 
 import { Step, Stepper, StepLabel } from 'material-ui/Stepper';
 
-class SpecTestContainer extends React.Component {
-  initialState = {
-    selectedSpec: '',
-    stepIndex: 0,
-    finished: false,
-    shouldDisplayWarningSnackBar: false,
-    items: [],
-    decks: [],
-    deckCount: [],
-    specName: '',
-    specNameErrorText: '',
-    value: 0,
-    selectedSpecContent: '',
-    specType: 'SpecTest'
-  };
-
-  initialBoardState = {
-    otherImages: [],
-    allImages: [],
-    allSpecs: [],
-    searchedSpec: [],
-    allElements: [],
-    standardElements: [],
-    toggableElements: [],
-    cardElements: [],
-    diceElements: [],
-    cardsDeckElements: [],
-    piecesDeckElements: [],
-    currentUserElements: [],
-    recentElements: [],
-    gameIcon50: [],
-    gameIcon512: [],
-    gameIcon50x50: constants.GAMEICON_50x50,
-    gameIcon512x512: constants.GAMEICON_512x512
-  };
-
-  initialVars = {
-    boardImage: '',
-    snackbarWarning: '',
-    boardSize: 0,
-    spec: [],
-    tutorialYoutubeVideo: constants.YOUTUBE_VIDEO,
-    wikipediaUrl: constants.WIKI_URL,
-    selectedUid: ''
-  };
-
-  state = Object.assign({}, this.initialState, this.initialBoardState);
-  vars = Object.assign({}, this.initialVars);
+interface SpecTestContainerProps {
+
+}
+
+interface SpecTestContainerState {
+  selectedSpec: string;
+  stepIndex: number;
+  finished: boolean;
+  shouldDisplayWarningSnackBar: boolean;
+  items: any;
+  decks: any;
+  deckCount: any;
+  specName: string;
+  specNameErrorText: string;
+  value: number;
+  selectedSpecContent: string;
+  specType: string;
+  otherImages: any;
+  allImages: any;
+  allSpecs: any;
+  searchedSpec: any;
+  allElements: any;
+  standardElements: any;
+  toggableElements: any;
+  cardElements: any;
+  diceElements: any;
+  cardsDeckElements: any;
+  piecesDeckElements: any;
+  currentUserElements: any;
+  recentElements: any;
+  gameIcon50: any;
+  gameIcon512: any;
+  gameIcon50x50: any;
+  gameIcon512x512: any;
+}
+
+class SpecTestContainer extends React.Component<SpecTestContainerProps, SpecTestContainerState> {
+
+  initialState: any;
```

Comment:
Same as before. So many `any` use. Though I said you can use `any`, I think these one's should be type tested as these should not be ever set to wrong type of value, as it will create a lot of errors.

Reply:
I've changed the type for other cases of Object.assign(), however, for this particular case the overall state is split up between initialState and initialBoardState. I could define two separate interfaces instead though.